ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.linear_rao;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.PstRange;
+import com.farao_community.farao.data.crac_api.RangeAction;
+import com.farao_community.farao.data.crac_result_extensions.*;
+import com.farao_community.farao.util.SystematicSensitivityAnalysisResult;
+import com.powsybl.iidm.network.Network;
+
+import java.util.*;
+
+/**
+ * An AbstractSituation includes a set of information associated to a given
+ * network situation.
+ *
+ * @author Philippe Edwards {@literal <philippe.edwards at rte-france.com>}
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public class Situation {
```

Comment:
Maybe situationManager is a better name for this class

Reply:
I don't know because I quite copied the Network methods concerning Situation so I imagined it as Network object. But I don't know. I feel like it's quite weird to have a SituationManager object but no Situation object

Reply:
I really don't like this name